tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/WcfTests.cs

```diff
@@ -23,55 +27,56 @@ public WcfTests(ITestOutputHelper output)
+            if (enableNewWcfInstrumentation)
+            {
+                SetEnvironmentVariable("DD_TRACE_WCF_ENABLE_NEW_INSTRUMENTATION", "true");
+            }
 
             Output.WriteLine("Starting WcfTests.SubmitsTraces. Starting the Samples.Wcf requires ADMIN privileges");
 
-            var expectedSpanCount = 4;
+            var expectedSpanCount = 9;
 
             const string expectedOperationName = "wcf.request";
-            const string expectedServiceName = "Samples.Wcf";
-            HashSet<string> expectedResourceNames = new HashSet<string>()
-            {
-                "http://schemas.xmlsoap.org/ws/2005/02/trust/RSTR/Issue",
-                "http://schemas.xmlsoap.org/ws/2005/02/trust/RST/Issue",
-                "http://schemas.xmlsoap.org/ws/2005/02/trust/RST/SCT",
-                "WcfSample/ICalculator/Add"
-            };
 
             int agentPort = TcpPortProvider.GetOpenPort();
             int wcfPort = 8585;
 
             using (var agent = new MockTracerAgent(agentPort))
-            using (RunSampleAndWaitForExit(agent.Port, arguments: $"WSHttpBinding Port={wcfPort}"))
+            using (RunSampleAndWaitForExit(agent.Port, arguments: $"{binding} Port={wcfPort}"))
             {
-                var spans = agent.WaitForSpans(expectedSpanCount, operationName: expectedOperationName);
-                Assert.True(spans.Count >= expectedSpanCount, $"Expecting at least {expectedSpanCount} spans, only received {spans.Count}");
-
-                foreach (var span in spans)
-                {
-                    // Validate server fields
-                    Assert.Equal(expectedServiceName, span.Service);
-                    Assert.Equal(ServiceVersion, span.Tags[Tags.Version]);
-                    Assert.Equal(expectedOperationName, span.Name);
-                    Assert.Equal(SpanTypes.Web, span.Type);
-                    Assert.Equal(SpanKinds.Server, span.Tags[Tags.SpanKind]);
+                var spans = agent.WaitForSpans(expectedSpanCount, operationName: expectedOperationName)
+                                    .Where(s => !s.Resource.Contains("schemas.xmlsoap.org") && !s.Resource.Contains("www.w3.org"));
 
-                    // Validate resource name
-                    Assert.Contains(span.Resource, expectedResourceNames);
+                var settings = VerifyHelper.GetSpanVerifierSettings(binding, enableCallTarget, enableNewWcfInstrumentation);
 
-                    // Test HTTP tags
-                    Assert.Equal("POST", span.Tags[Tags.HttpMethod]);
-                    Assert.Equal("http://localhost:8585/WcfSample/CalculatorService", span.Tags[Tags.HttpUrl]);
-                    Assert.Equal($"localhost:{wcfPort}", span.Tags[Tags.HttpRequestHeadersHost]);
-                }
+                // Overriding the type name here as we have multiple test classes in the file
+                // Ensures that we get nice file nesting in Solution Explorer
```

Comment:
Is that true? I only see one class 😄

Reply:
Heh silly copy/paste error 🙃 Fixed in a9fabf6a290f181621790487cdb34173d5b3292f